updateHostsFile.py

```diff
@@ -438,14 +438,35 @@ def updateReadmeData():
 
 
 def moveHostsFileIntoPlace(finalFile):
+    """
+    Move the newly-created hosts file into its correct location on the OS.
+
+    For UNIX systems, the hosts file is "etc/hosts." On Windows, it's
+    "C:\Windows\system32\drivers\etc\hosts."
+
+    For this move to work, you must have administrator privileges to do this.
+    On UNIX systems, this means having "sudo" access, and on Windows, it
+    means being able to run command prompt in administrator mode.
+
+    Parameters
+    ----------
+    finalFile : str
+        The name of the newly-created hosts file to move.
+    """
+
+    filename = os.path.abspath(finalFile.name)
+
     if os.name == "posix":
-        print ("Moving the file requires administrative privileges. " +
-               "You might need to enter your password.")
-        if subprocess.call(["/usr/bin/sudo", "cp", os.path.abspath(finalFile.name), "/etc/hosts"]):
+        print("Moving the file requires administrative privileges. "
+              "You might need to enter your password.")
+        if subprocess.call(["/usr/bin/sudo", "cp", filename, "/etc/hosts"]):
             printFailure("Moving the file failed.")
     elif os.name == "nt":
-        print("Automatically moving the hosts file in place is not yet supported.")
-        print("Please move the generated file to %SystemRoot%\system32\drivers\etc\hosts")
+        print("Moving the file requires administrative "
+              "privileges in command prompt.")
+        if subprocess.call(["cmd", "/C", "copy", filename,
+                            "C:\Windows\system32\drivers\etc\hosts"]):
+            printFailure("Moving the file failed.")
 
```

Comment:
Hmm...to be fair, you do advocate using the batch files in the `README`, which does in fact copy files directly into this directory (you can't run this command unless you're administrator).  Thus, I'm not so concerned about that user's rights (your second point).

In terms of Windows compatibility across versions, I'm not sure what the target is, but I know that it works with machines as old as Windows 7.  I figured that unless someone has a machine older than this, that won't be a problem.

As for your final point, I do agree that this is a problem.  However, if people do run into this problem, we can always tell people to move the file manually unless they can provide a nice solution in code.

In short, I don't believe that that will be the case.  Let me know what you think.